Model/Api/UpdateCart.php

```diff
@@ -203,6 +203,19 @@ public function execute($cart, $add_items = null, $remove_items = null, $discoun
                         // Already sent a response with error, so just return.
                         return false;
                     }
+                    // Adding the same data into immutable quote
+                    // in order to avoid missing addons in parent quote
+                    $addToImmutableQuote = true;
+                    if ($immutableQuote->getItems()) {
+                        $addToImmutableQuote = false;
```

Comment:
Hmm, why do we check this? quote always has items, right?

Reply:
Yeah, you're right
I've replaced the old correct validation with this one in a hurry, will rollback this change in a moment